sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusReceiverClientIntegrationTest.java

```diff
@@ -0,0 +1,578 @@
+                logger.warning("Error occurred when draining deferred messages Entity : " + receiver.getEntityPath(), e);
+            }
+
+            // set empty list
+            messagesDeferred.set(new ArrayList<>());
+        }
+
+        dispose(receiver, sender, receiveAndDeleteReceiver);
+    }
+
+    /**
+     * Verifies that we can send and receive two messages.
+     */
+    @MethodSource("messagingEntityWithSessions")
+    @ParameterizedTest
+    void receiveTwoMessagesAutoComplete(MessagingEntityType entityType, boolean isSessionEnabled) {
+        // Arrange
+        setSenderAndReceiver(entityType, isSessionEnabled);
+        int howManyMessage = 2;
+
+        final String messageId = UUID.randomUUID().toString();
+        final ServiceBusMessage message = getMessage(messageId, isSessionEnabled);
+
+        sendMessage(message);
+        sendMessage(message);
+
+        // Act
+        Iterable<ServiceBusReceivedMessage> iterableMessages = receiver.receive(howManyMessage, TIMEOUT);
+
+        // Assert
+        for (ServiceBusReceivedMessage receivedMessage: iterableMessages) {
+            assertMessageEquals(receivedMessage, messageId, isSessionEnabled);
+        }
+
+        messagesPending.decrementAndGet();
+        messagesPending.decrementAndGet();
+    }
+
+    /**
+     * Verifies that we do not receive any message in given timeout.
+     */
+    @MethodSource("messagingEntityWithSessions")
+    @ParameterizedTest
+    void receiveNoMessage(MessagingEntityType entityType, boolean isSessionEnabled) {
+        // Arrange
+        setSenderAndReceiver(entityType, isSessionEnabled);
+        int howManyMessage = 2;
+        int noMessages = 0;
+
+        // Act
+        IterableStream<ServiceBusReceivedMessage> iterableMessages = receiver.receive(howManyMessage, Duration.ofSeconds(15));
+
+        // Assert
+        final int receivedMessages = iterableMessages.stream().collect(Collectors.toList()).size();
+        assertEquals(noMessages, receivedMessages);
+    }
+
+    /**
+     * Verifies that we can send and receive one messages.
+     */
+    @MethodSource("messagingEntityWithSessions")
+    @ParameterizedTest
+    void receiveMessageAutoComplete(MessagingEntityType entityType, boolean isSessionEnabled) {
+        // Arrange
+        setSenderAndReceiver(entityType, isSessionEnabled);
+        int howManyMessage = 1;
+
+        final String messageId = UUID.randomUUID().toString();
+        final ServiceBusMessage message = getMessage(messageId, isSessionEnabled);
+
+        sendMessage(message);
+
+        // Act
+        Iterable<ServiceBusReceivedMessage> iterableMessages = receiver.receive(howManyMessage, TIMEOUT);
+
+        // Assert
+        for (ServiceBusReceivedMessage receivedMessage: iterableMessages) {
+            assertMessageEquals(receivedMessage, messageId, isSessionEnabled);
+        }
+
+        messagesPending.decrementAndGet();
+        messagesPending.decrementAndGet();
```

Comment:
You only received one message and sent one message, this should be decremented once.

Reply:
This should be in the foreach loop as well.